src/Test/L0/Worker/SetVariableRestrictionsL0.cs

```diff
@@ -0,0 +1,259 @@
+                Assert.Throws<InvalidOperationException>(() => new TaskSetVariableCommand().Execute(_ec.Object, command));
+            }
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void ExactMatchAllowed()
+        {
+            using (TestHostContext hc = CreateTestContext())
+            {
+                var restrictions = new TaskRestrictions() { SettableVariables = new TaskVariableRestrictions() };
+                restrictions.SettableVariables.Allowed.Add("myVar");
+                restrictions.SettableVariables.Allowed.Add("otherVar");
+                _ec.Object.Restrictions.Add(restrictions);
+
+                TaskSetVariableCommand setVariable;
+                Command command;
+                var value = "myValue";
+
+                foreach(String variable in new String[] { "myVar", "myvar", "MYVAR", "otherVAR" })
+                {
+                    command = SetVariableCommand(variable, value);
+                    setVariable = new TaskSetVariableCommand();
+                    setVariable.Execute(_ec.Object, command);
+                    Assert.Equal(value, _variables.Get(variable));
+                }
+
+                var badVar = "badVar";
+                command = SetVariableCommand(badVar, value);
+                setVariable = new TaskSetVariableCommand();
+                Assert.Throws<InvalidOperationException>(() => setVariable.Execute(_ec.Object, command));
+            }
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void MiniMatchAllowed()
+        {
+            using (TestHostContext hc = CreateTestContext())
+            {
+                var restrictions = new TaskRestrictions() { SettableVariables = new TaskVariableRestrictions() };
+                restrictions.SettableVariables.Allowed.Add("my*");
+                _ec.Object.Restrictions.Add(restrictions);
+
+                TaskSetVariableCommand setVariable;
+                Command command;
+                var value = "myValue";
+
+                foreach(String variable in new String[] { "myVar", "mything", "MY" })
+                {
+                    command = SetVariableCommand(variable, value);
+                    setVariable = new TaskSetVariableCommand();
+                    setVariable.Execute(_ec.Object, command);
+                    Assert.Equal(value, _variables.Get(variable));
+                }
+
+                var badVar = "badVar";
+                command = SetVariableCommand(badVar, value);
+                setVariable = new TaskSetVariableCommand();
+                Assert.Throws<InvalidOperationException>(() => setVariable.Execute(_ec.Object, command));
+            }
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void MultipleRestrictionsMostRestrictive()
```

Comment:
Love seeing design decisions turned into unit tests like this - overall, this coverage is excellent